<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <!-- Topics parameters-->
    <param name="strImage_sub_topic"        value="/rgb/image" />
    <param name="strImage_pub_topic"        value="/blob_detection/image_blob"/>
    <param name="strMask_pub_topic"         value="/blob_detection/image_mask" />
    <param name="strAgimusFrame_sub_topic"  value="/agimus/vision/tags" />
    <param name="strCameraInfo_sub_topic"   value="/rgb/camera_info" />
    <param name="strTransform_sub_topic"    value="/agimus/vision/tags" />

    <!-- Parameters for HSV mask filter -->
    <param name="iHSV_min_H" value = "0"/>
    <param name="iHSV_min_S" value = "0"/>
    <param name="iHSV_min_V" value = "0"/>
    <param name="iHSV_max_H" value = "255"/>
    <param name="iHSV_max_S" value = "255"/>
    <param name="iHSV_max_V" value = "25"/>

    <!-- Parameters for blob properties-->
    <param name="fMinThreshold"         value = "0.0"/>
    <param name="fMaxThreshold"         value = "100.0"/>
    <param name="bFilterByArea"         value = "true"/>
    <param name="fMinArea"              value = "2.0"/>
    <param name="fMaxArea"              value = "2000.0"/>
    <param name="bFilterByCircularity"  value = "true"/>
    <param name="fMinCircularity"       value = "0.89"/>
    <param name="fMaxCircularity"       value = "1.0"/>
    <param name="bFilterByConvexity"    value = "true"/>
    <param name="fMinConvexity"         value = "0.2"/>
    <param name="fMaxConvexity"         value = "1.0"/>
    <param name="bFilterByInertia"      value = "true"/>
    <param name="fMinInertiaRatio"      value = "0.3"/>
    <param name="fMaxInertiaRatio"      value = "1.0"/>


    <node name="blob_detection" pkg="tiago_blob_detection" type="blob_detection" output ="screen"/>
</launch>